compiler/infra/ForwardList.hpp

```diff
@@ -0,0 +1,684 @@
+#ifndef FORWARD_LIST_HPP
```

Comment:
I didn't realize deque.hpp had been lower-cased.  The previous example that I followed was ReferenceWrapper.hpp do we want to open an issue to rename it to reference_wrapper.hpp?